File: Prob933/Solution.java
// Approach 1:-

class RecentCounter {
    
    private Queue<Integer> q;

    public RecentCounter() {
        q = new LinkedList<Integer>();
    }
    
    public int ping(int t) {
        q.add(t);
        
        while((t-q.peek())>3000)
            q.poll();
        
        return q.size();
    }
}

File: Prob933/Solution.cpp
// Approach 1:-

class RecentCounter
{
private:
    queue<int> q;

public:
    RecentCounter()
    {
    }

    int ping(int t)
    {
        q.push(t);

        while ((t - q.front()) > 3000)
            q.pop();

        return q.size();
    }
};